edit/security_functions.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ * Verifies the username and password. If it is correct, any session data relating to incorrect entry is removed.
+ *
+ * @param string $username The given username.
+ * @param string $password The given password.
+ *
+ * @return bool Returns TRUE if the credentials match, otherwise FALSE.
+ */
+function verifyLogin($username, $password) : bool {
+    session_start();
+    if (loginExists($username)) {
+        if (correctPassword($username, $password)) {
+            unset($_SESSION['badname']);
+            unset($_SESSION['badpass']);
+            return true;
+        }
+    }
+    return false;
+}
+
+/*
+ * If the cookie ID does not match the session ID, the user is punted out of the session. GET OUTTA HERE
+ */
+function verifyUser() {
+    session_start();
+    if ($_COOKIE['id'] != session_id()) {
+        setcookie('id',0, time() - 1);
+        header('Location: index.php');
+    }
+    session_destroy();
+}
+
+/*
+ * Verifies the username and password. If it is correct, any session data relating to incorrect entry is removed.
+ *
+ * @return string Returns a message if either the username or password is incorrect. If the user has entered no data,
+ * this value is blank.
+ */
+function loginFailMessage() : string {
+    session_start();
+    if (array_key_exists('badname' , $_SESSION)) {
+        session_destroy();
+        return 'User not in database. Please try again';
```

Comment:
by telling the user if they have a incorrect user or password you are opening up yourself to a brute force attack, people can just keep trying usernames until the error doesnt appear. should show a generic error that says incorrect username or password. poorer UX but better security